narwhals/_arrow/series.py

```diff
@@ -780,6 +780,56 @@ def convert_time_zone(self: Self, time_zone: str) -> ArrowSeries:
 
         return self._arrow_series._from_native_series(result)
 
+    def timestamp(self: Self, time_unit: Literal["ns", "us", "ms"] = "us") -> ArrowSeries:
+        import pyarrow as pa  # ignore-banned-import
+        import pyarrow.compute as pc  # ignore-banned-import
+
+        s = self._arrow_series._native_series
+        dtype = self._arrow_series.dtype
+        if dtype == self._arrow_series._dtypes.Datetime:
+            unit = dtype.time_unit  # type: ignore[attr-defined]
+            s_cast = s.cast(pa.int64())
+            if unit == "ns":
+                if time_unit == "ns":
+                    result = s_cast
+                if time_unit == "us":
+                    result = pc.divide(s_cast, 1_000)
+                if time_unit == "ms":
+                    result = pc.divide(s_cast, 1_000_000)
```

Comment:
for all of these, it would be marginally more efficient to use `elif`, else we're doing similar comparisons

we're already validating `time_unit` in `narwhals/expr.py`, so this could just be
```
                if time_unit == "ns":
                    result = pc.multiply(s_cast, 1_000)
                elif time_unit == "us":
                    result = s_cast
                else:
                    result = pc.divide(s_cast, 1_000)
```